report_aeroo/report_aeroo.py

```diff
@@ -182,23 +194,52 @@ def create(self, cr, uid, ids, data, context=None):
         else:
             context = dict(context)
 
+        env = api.Environment(cr, uid, context)
+
+        name = self.name.startswith('report.') and self.name[7:] or self.name
+
+        report_xml = env['ir.actions.report.xml'].search(
+            [('report_name', '=', name)])
+
+        output_format = report_xml.out_format.code[3:]
+
         if len(ids) > 1:
-            raise ValidationError(
-                _('Aeroo Reports do not support generating reports in batch. '
-                  'You must select one record at a time.'))
+            if output_format != 'pdf':
+                raise ValidationError(
+                    _('Aeroo Reports do not support generating non-pdf '
+                      'reports in batch. You must select one record '
+                      'at a time.'))
 
-        env = api.Environment(cr, uid, context)
+            file_names = []
+
+            for record_id in ids:
+                report = self.create(cr, uid, [record_id], data, context)
+
+                temp_file = self.generate_temporary_file(
+                    cr, uid, [record_id], report[0], output_format)
+                file_name = temp_file.name[:-3] + output_format
+                file_names.append(file_name)
+
+            output_file = NamedTemporaryFile(
+                suffix='.%s' % output_format, delete=False)
+            output_file.close()
+
+            output_filename = output_file.name[:-3] + output_format
+            args = ['pdftk']
+            args += file_names
+            args += ['cat', 'output', output_filename]
+            subprocess.Popen(args)
+            time.sleep(0.1)
```

Comment:
@yasmineelmrini this will not always work. If your process gets stuck (and it will happen), you get a zombie process that will never be killed.

You must create a while loop and after some delay kill the process. (see libreoffice_timeout for an example). Since the code will be the same as libreoffice, you could isolate the subprocess part inside a common function.